community/cypher/cypher/src/main/scala/org/neo4j/cypher/internal/spi/v2_2/TransactionBoundQueryContext.scala

```diff
@@ -131,10 +132,13 @@ final class TransactionBoundQueryContext(graph: GraphDatabaseAPI,
     case Some(typeIds) => JavaConversionSupport.asScala(statement.readOperations().nodeGetRelationships(node.getId, dir, typeIds: _* )).map(relationshipOps.getById)
   }
 
-  def exactIndexSearch(index: IndexDescriptor, value: Any) =
+  def exactIndexSearch(index: IndexDescriptor, value: Any) = {
+    indexSearchMonitor.exactIndexSearch(index, value)
     mapToScala(statement.readOperations().nodesGetFromIndexLookup(index, value))(nodeOps.getById)
+  }
 
-  def exactUniqueIndexSearch(index: IndexDescriptor, value: Any): Option[Node] = {
+  def lockingIndexSearch(index: IndexDescriptor, value: Any): Option[Node] = {
```

Comment:
Naming of these methods is a bit inconsistent. Maybe call this one `lockingExactIndexSearch`?
Or maybe we can name them `exactIndexSearchForRead` and `exactIndexSearchForWrite`.